cmd/chartsvc/handler.go

```diff
@@ -444,3 +445,75 @@ func newChartVersionListResponse(c *models.Chart) apiListResponse {
 
 	return cvl
 }
+
+func handleRedirect(w http.ResponseWriter, req *http.Request) {
+
+	// The URL can be in one of two forms:
+	// - /v1/redirect/charts/stable/aerospike
+	// - /v1/redirect/charts/stable/aerospike/v1.2.3
+	// And either of these can optionally have a trailing /
+
+	// Make sure the path is valid
+	ct := strings.TrimPrefix(req.URL.Path, "/v1/redirect/charts/")
+
+	// check if URL for provenance
+	prov := strings.HasSuffix(ct, ".prov")
```

Comment:
I think we may also want to consider handling the provenance file in a separate handler

Reply:
I'm not sure how to handle that with mux (see `setupRoutes`). `PathPrefix` is used because the redirect URL is going to have multiple path parts to it. In the router there is not method for filtering based on suffix, that I'm aware of. Do you have a suggestion?